### PR TITLE
feat: add universal2 support

### LINKED ISSUE #35: Wrong architecture for ninja binary on Apple Silicon
Currently pip only provides a x86_64 binary on Apple Silicon, leading to a version of ninja on the system which can fail to build correct targets (see https://gitlab.kitware.com/cmake/cmake/-/issues/20966 for reference).

The pip should provide an universal binary instead, or a platform specific one. Initial tests show both approaches would resolve the issue.

### PR DESCRIPTION
**Pull request by mayeut**
Fixes #35 & #41 
Depends on #61 & #62

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
-BUILD_ENV = {
-    "MACOSX_DEPLOYMENT_TARGET": "10.9",
-}
-
+if sys.platform.startswith("darwin"):
+    BUILD_ENV = {
+        "MACOSX_DEPLOYMENT_TARGET": "10.9",
+        "CMAKE_OSX_ARCHITECTURES": "arm64;x86_64",
+        "CFLAGS": "-save-temps",
+        "CXXFLAGS": "-save-temps",
```
Comment (mayeut) [inline noxfile.py:15]: Not sure why this is required when building locally but not in GHA...
It definitely fails without this on my laptop.

File: pyproject.toml [hunk 2]
```diff
@@ -19,6 +19,7 @@ manylinux-i686-image = "manylinux1"
 
 [tool.cibuildwheel.macos.environment]
 MACOSX_DEPLOYMENT_TARGET = "10.9"
+CMAKE_OSX_ARCHITECTURES = "arm64;x86_64"
```
Comment (mayeut) [inline pyproject.toml:22]: We should be able to get rid of this once scikit-build 0.12 lands.

**Comment by henryiii** [comment]
I'll push a cmake 3.21.1 release, @mayeut or @jcfr would you like to push a ninja release?

**Comment by mayeut** [comment]
I'll push one for Ninja

**Pull request merged**